extensions/http-curl/client/HTTPClient.cpp

```diff
@@ -328,18 +325,15 @@ void HTTPClient::setRequestHeader(std::string key, std::optional<std::string> va
 }
 
 namespace {
-struct CurlSListFreeAll {
-  void operator()(struct curl_slist* slist) const {
-    curl_slist_free_all(slist);
-  }
-};
+using CurlSlistDeleter = decltype([](struct curl_slist* slist) { curl_slist_free_all(slist); });
 
-std::unique_ptr<struct curl_slist, CurlSListFreeAll> getCurlSList(const std::unordered_map<std::string, std::string>& request_headers) {
+std::unique_ptr<struct curl_slist, CurlSlistDeleter> toCurlSlist(const std::unordered_map<std::string, std::string>& request_headers) {
   curl_slist* new_list = nullptr;
+  const auto guard = gsl::finally([&new_list]() { curl_slist_free_all(new_list); });
```

Comment:
this could be
```suggestion
  const auto guard = gsl::finally([&new_list]() { CurlSListDeleter()(new_list); });
```
to avoid duplication of the logic

Reply:
I think it would improve readability: "use `CurlSListDeleter` on `new_list` if there is an exception, otherwise transfer ownership to the return value, which also uses `CurlSListDeleter`" -- instead of "use `curl_slist_free_all` or `CurlSListDeleter` depending on the situation", and then one has to do an extra step to find out that those two are the same.

But I agree it doesn't matter much either way.

Reply:
`CurlSListDeleter` is a type whose whole purpose is to call `curl_slist_free_all` as a `unique_ptr` deleter. I can add a comment to make that clear if this helps the case, but I prefer not to use it for anything other than a `unique_ptr` deleter for `curl_slist`.